clustered/common/src/main/java/org/ehcache/clustered/common/store/ElementBuilder.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.common.store;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Builds {@link Element}s
+ */
+public interface ElementBuilder {
+
+  /**
+   * Gets an updated {@link Element} from existing element
+   * with new payLoad
+   *
+   * @param element the old element
+   * @param payLoad the new payLoad
+   * @return the Element with a new payLoad
+   */
+  Element getElement(Element element, ByteBuffer payLoad);
```

Comment:
@ljacomet  that's the question I am asking, do we say that client only sends a payload in form of dummy element, because Element is a wrapper around payload with some more info in to it for Chain. If we got to compare two Chains at the server side, there has to be some identifier in implementation to enable that. 
So the current method provides possibility to copy that extra info(otherwise not available at client side) in the new element with the payload.